Implement KiwiTalk login screen

storycraft opened the pull request:
키위톡 로그인 창과 로그인 기능을 구현합니다.

## 구현된 것
* [x] 계정 로그인
* [x] 기기 등록
* [x] 에러 메세지

This pr close #1259

storycraft committed:
Fix LoginForm to use form

storycraft committed:
Fix WindowTitleBar to use tauri attribute

storycraft committed:
Remove window decorations and adjust default size

storycraft committed:
Removed email type as phone number can be used to login

storycraft committed:
Remove font-awesome script

storycraft committed:
Add inital app login

storycraft committed:
Merge branch 'add-intl' into add-app-login

storycraft committed:
Merge branch 'dev' into add-app-login

storycraft committed:
Add AppWindowControl

storycraft committed:
Use form on LoginForm

storycraft committed:
Prevent form submit

storycraft committed:
Split login page content sections

storycraft committed:
Add simple error message

storycraft committed:
Do not show error message when registeration required

storycraft committed:
Add onLogin prop to AppLogin

storycraft committed:
Add reset button.
Merge errorMessage state.

storycraft committed:
Fix state sync

storycraft committed:
Refactor state type

storycraft committed:
Add login status translation(ko-KR)

storycraft commented:
This pr close #1259

RanolP commented on frontend/src/app/components/login/form/login.tsx:
```diff
@@ -60,10 +60,9 @@ export const LoginForm = ({
   }
 
-  return <div className={className}>
+  return <form className={className} onSubmit={(e) => e.preventDefault()}>
```
1. 현재 코드를 보면 Controlled Component\*일 이유가 적은 것 같습니다. [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData#prepopulating_from_a_html_form_element) API를 활용해 Uncontrolled Component\*으로 만드는 것도 괜찮아 보입니다. 필요에 따라 [react-hook-form](https://www.npmjs.com/package/react-hook-form) 등의 라이브러리를 고려할 수 있습니다. 만약 추후 실시간 검증 등의 기능이 도입될 예정이라면 Controlled Component를 잘 쓰기 위해 [formik](https://www.npmjs.com/package/formik)의 도입을 고려할 수 있습니다.
2. `<LoginButton>`의 `onClick`과 함께 사용하는 대신 `onSubmit`에서 함께 처리하는 편이 좋겠습니다.

*[Controlled vs. Uncontrolled Components](https://reactjs.org/docs/glossary.html#controlled-vs-uncontrolled-components), 다만 저는 Controlled Component를 선호하라는 설명에 동의하지 않습니다.

storycraft replied:
1. 현재 로그인 폼은 Controlled Component가 아닙니다
2. 이게 좋을 것 같습니다

storycraft replied:
만약 FormData를 사용하는 것으로 바꾸게 된다면 현재 사용하고 있는 input 텍스트와 체크박스 컴포넌트 prop에 name이 없기 때문에 이쪽도 같이 수정합니다

storycraft replied:
FormData로 전환에 큰 문제가 없어서 바로 적용했습니다

storycraft committed:
Remove newline

storycraft committed:
Add AsyncLock hook

storycraft committed:
Apply AsyncLock hook to login contents

storycraft committed:
Fix to lock immediately after acquire

storycraft committed:
Rename type name

storycraft committed:
Use submit event

storycraft committed:
Rename to defaultInput

storycraft committed:
Remove id prop from CheckBox.
Add name prop to CheckBox, InputForm.
Wrap with label insteadof div on input components.

storycraft committed:
Use FormData

storycraft committed:
Fix inner checkbox container display to flex
